cypress/e2e/signUp.cy.js

```diff
@@ -1,12 +1,71 @@
 /// <reference types='cypress' />
 /// <reference types='../support' />
+import SignUpPage from '../support/pages/signUp.pageObject';
+import HomePageObject from '../support/pages/home.pageObject';
+import faker from 'faker';
 
 describe('Sign Up page', () => {
+  let name;
+  let email;
+  let password;
+  const signUpPage = new SignUpPage();
+
   before(() => {
+    cy.task('db:clear');
+  });
+
+  beforeEach(() => {
+    cy.visit('/#/register');
+    name = faker.name.firstName();
+    email = faker.internet.email();
+    password = faker.internet.password();
+  });
+
+  it('should create a new user', () => {
+    signUpPage.fillName(name);
+    signUpPage.fillEmail(email);
+    signUpPage.fillPassword(password);
+
+    signUpPage.submitSignUpForm();
+
+    cy.contains('Welcome!');
+    cy.contains('Your registration was successful!');
+    cy.contains('OK').click();
+
+    const homePage = new HomePageObject();
+    homePage.assertHeaderContainUsername(name);
+  });
+
```

Comment:
All tests are currently running independently from one another. It's a good practice in e2e testing to have some tests depend on the outcome of previous tests. For example, you might want to test logging in with the user that was created in the first test.

Reply:
done

Reply:
done